nixpkgs-overlays/hack-get/default.nix

```diff
@@ -8,27 +8,56 @@ self:
   # Retrieve source that is controlled by the hack-* scripts; it may be either a stub or a checked-out git repo
   hackGet = p:
     let
+      contents = builtins.readDir p;
+
+      contentsMatch = { required, optional }:
+           (let all = required // optional; in all // contents == all)
+        && builtins.intersectAttrs required contents == required;
+
+      # Newer obelisk thunks include the feature of hackGet with a src.nix file in the thunk.
+      isObeliskThunkWithSrc =
+        let
+          common = {
+            required = { "default.nix" = "regular"; "src.nix" = "regular"; };
+            optional = { ".attr-cache" = "directory"; };
+          };
+          packed = jsonFileName: {
+            required = { ${jsonFileName} = "regular"; } // common.required;
+            optional = common.optional;
+          };
+          unpacked = {
+            required = { "unpacked" = "directory"; } // common.required;
+            optional = common.optional;
+          };
+        in contentsMatch unpacked
+          || builtins.any (n: contentsMatch (packed n)) [ "git.json" "github.json" ];
+
       filterArgs = x: removeAttrs x [ "branch" ];
       hasValidThunk = name: if builtins.pathExists (p + ("/" + name))
-        then (let contents = builtins.readDir p;
-              in if contents == { ${name} = "regular"; } ||
-                    contents == { ${name} = "regular"; ".attr-cache" = "directory"; } ||
-                    contents == { ${name} = "regular"; "default.nix" = "regular"; } ||
-                    contents == { ${name} = "regular"; "default.nix" = "regular"; ".attr-cache" = "directory"; }
-                 then true
-                 else throw "Thunk at ${toString p} has files in addition to ${name} and optionally default.nix and .attr-cache. Remove either ${name} or those other files to continue.")
+        then
+          contentsMatch {
+            required = { ${name} = "regular"; };
+            optional = { "default.nix" = "regular"; ".attr-cache" = "directory"; };
+          }
+          || throw "Thunk at ${toString p} has files in addition to ${name} and optionally default.nix and .attr-cache. Remove either ${name} or those other files to continue (check for leftover .git too)."
         else false;
-    in if hasValidThunk "git.json" then (
-      let gitArgs = filterArgs (builtins.fromJSON (builtins.readFile (p + "/git.json")));
-      in if builtins.elem "@" (lib.stringToCharacters gitArgs.url)
-      then self.fetchgitPrivate gitArgs
-      else self.fetchgit gitArgs)
-    else if hasValidThunk "github.json" then
-      self.fetchFromGitHub (filterArgs (builtins.fromJSON (builtins.readFile (p + "/github.json"))))
-    else {
-      name = baseNameOf p;
-      outPath = self.filterGit p;
-    };
+    in
+      if isObeliskThunkWithSrc then
+        builtins.trace
+          "DEPRECATED: hackGet is deprecated for thunks containing src.nix; Use `import ${p}/src.nix` instead of `hackGet ${p}`"
```

Comment:
```suggestion
          "DEPRECATED: hackGet is deprecated for thunks NOT containing src.nix; Use `import ${p}/src.nix` instead of `hackGet ${p}`"
```